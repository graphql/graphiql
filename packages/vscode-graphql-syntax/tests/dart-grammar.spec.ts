import { tokenizeFile } from './__utilities__/utilities';

describe('inline.graphql.dart grammar', () => {
  const scope = 'inline.graphql.dart';

  it('should tokenize a simple dart file', async () => {
    const result = await tokenizeFile('__fixtures__/test.dart', scope);
    expect(result).toMatchSnapshot();
  });
});
