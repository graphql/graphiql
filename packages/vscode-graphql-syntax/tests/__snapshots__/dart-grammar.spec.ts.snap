// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`inline.graphql.dart grammar > should tokenize a simple dart file 1`] = `
const query1 =  | 
'''             | string.interpolated.triple.single.dart
#graphql        | comment.line.graphql.js
                | 
query           | keyword.operation.graphql
                | meta.selectionset.graphql
{               | meta.selectionset.graphql punctuation.operation.graphql
                | meta.selectionset.graphql
id              | meta.selectionset.graphql variable.graphql
                | meta.selectionset.graphql
}               | meta.selectionset.graphql punctuation.operation.graphql
'''             | string.interpolated.triple.single.dart
;               | 
                | 
const query2 =  | 
"""             | string.interpolated.triple.double.dart
#graphql        | comment.line.graphql.js
                | 
query           | keyword.operation.graphql
                | meta.selectionset.graphql
{               | meta.selectionset.graphql punctuation.operation.graphql
                | meta.selectionset.graphql
id              | meta.selectionset.graphql variable.graphql
                | meta.selectionset.graphql
}               | meta.selectionset.graphql punctuation.operation.graphql
"""             | string.interpolated.triple.double.dart
;               | 
                | 
`;
